scapy/layers/dns.py

```diff
@@ -240,6 +240,8 @@ def field_gen(dns_pkt):
                 continue
             for current in lay:
                 for field in current.fields_desc:
+                    # 39 isn't compressed because the DNAME RDATA target
+                    # name MUST NOT be sent out in compressed form
```

Comment:
Having thought about this I'm inclined to say that I'd probably compress it too to make it consistent with all the other RRs. It should make it easier to test various code paths in DNS-related software that is supposed to handle uncompressed and compressed RRs (regradless of whether they should be compressed or not according to RFCs).

Either way to me the most important part of this PR is that it's possible to use
```
DNSRR(..., rdata='local')
```
instead of
```
DNSRR(..., rdata=b'\x05local\x00')
```
to create valid DNAME RRs. I can leave the compression part as is or just add `39` to the list below to make it compressable.

Reply:
Honnestly this is your call. Looks good to me in both cases.

Reply:
Feel free to resolve to merge if you want.

Reply:
Sorry. I decided to compress it. I've just updated the test and I'll force-push it shortly.